cpp/src/gandiva/tree_expr_builder.cc

```diff
@@ -190,6 +190,12 @@ ConditionPtr TreeExprBuilder::MakeCondition(const std::string& function,
   return ConditionPtr(new Condition(func_node));
 }
 
+NodePtr TreeExprBuilder::MakeInExpressionDecimal(NodePtr node,
+                                                 const std::unordered_set<arrow::Decimal128> &constants,
```

Comment:
i think it is better to use gandiva::DecimalScalar128 instead of arrow::Decimal128 since in gandiva we are using gandiva::DecimalScalar128 as literal

Reply:
Substituted